src/native/libs/System.Native/pal_io.c

```diff
@@ -1883,6 +1883,53 @@ int32_t SystemNative_PWrite(intptr_t fd, void* buffer, int32_t bufferSize, int64
     return (int32_t)count;
 }
 
+#if (HAVE_PREADV || HAVE_PWRITEV) && !defined(TARGET_WASM)
+static int GetAllowedVectorCount(IOVector* vectors, int32_t vectorCount)
+{
+#if defined(IOV_MAX)
+    const int IovMax = IOV_MAX;
+#else
+    // In theory all the platforms that we support define IOV_MAX,
```

Comment:
Should we just fail to build if it's not defined?

Reply:
> Should we just fail to build if it's not defined?

We could end up breaking some configuration that is not maintained/built by us, which is something that I would prefer to avoid if possible.